src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -101,22 +101,27 @@
 
     @POST
     @RolesAllowed("write")
-    public Uni<Endpoint> createEndpoint(@Context SecurityContext sec, @NotNull @Valid Endpoint endpoint) {
+    @APIResponse(responseCode = "201", content = @Content(schema = @Schema(implementation = Endpoint.class)))
+    public Uni<Response> createEndpoint(@Context SecurityContext sec, @NotNull @Valid Endpoint endpoint) {
         RhIdPrincipal principal = (RhIdPrincipal) sec.getUserPrincipal();
         endpoint.setTenant(principal.getAccount());
 
+        Uni<Endpoint> endpointUni;
+
         if (endpoint.getType() != Endpoint.EndpointType.DEFAULT && endpoint.getProperties() == null) {
             throw new BadRequestException("Properties is required");
         } else if (endpoint.getType() == Endpoint.EndpointType.DEFAULT) {
             // Only a single default endpoint is allowed
-            return resources.getEndpointsPerType(principal.getAccount(), Endpoint.EndpointType.DEFAULT, false, null)
+            endpointUni = resources.getEndpointsPerType(principal.getAccount(), Endpoint.EndpointType.DEFAULT, false, null)
                     .toUni()
                     .onItem()
                     .ifNull()
                     .switchTo(resources.createEndpoint(endpoint));
+        } else {
+            endpointUni = resources.createEndpoint(endpoint);
         }
 
-        return resources.createEndpoint(endpoint);
+        return endpointUni.onItem().transform(e -> Response.created(null).entity(e).build());
```

Comment:
I wonder if the single letter `e` could be confused with "Exception" or "Error". I often use it in `... catch (Exception e) ...`

Reply:
That's true. I'll update.

Reply:
Also when I look up the semantics of a 201 I am not sure that matches. See e.g. 
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

Reply:
mmh.. True. It says that I should return an URL.

Reply:
To be fair, I don't see anywhere that we MUST use `201` when creating resources instead of `200`.

Reply:
This is written here:

https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html (section `9.5 POST`)

> If a resource has been created on the origin server, the response SHOULD be 201 (Created) and contain an entity which describes the status of the request and refers to the new resource, and a Location header (see section 14.30).

Reply:
It says SHOULD. Not MUST :-)

And iirc JavaScript clients seem to have larger issues looking at headers, which is why paging is not via header-fields (`X-href rel=next`), but wrapping the real payload.